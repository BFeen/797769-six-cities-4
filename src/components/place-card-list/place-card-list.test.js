import React from "react";
import renderer from "react-test-renderer";
import PlaceCardList from "./place-card-list.jsx";


const Offers = [
  {
    id: 0,
    title: `Beautiful & luxurious apartment at great location`,
    picture: `img/apartment-01.jpg`,
    isPremium: true,
    price: 200,
    type: `Apartment`,
    rating: 4,
  }, {
    id: 1,
    title: `Wood and stone place`,
    picture: `img/studio-01.jpg`,
    isPremium: true,
    price: 170,
    type: `House`,
    rating: 5,
  }, {
    id: 2,
    title: `Canal view Princengracht`,
    picture: `img/room.jpg`,
    isPremium: false,
    price: 70,
    type: `Room`,
    rating: 3,
  }, {
    id: 3,
    title: `Nice, cozy, warm big bed apartment`,
    picture: `img/apartment-02.jpg`,
    isPremium: false,
    price: 150,
    type: `Apartment`,
    rating: 4,
  }
];

describe(`PlaceCardList snapshot checking`, () => {
  it(`PlaceCardList rendering`, () => {
    const tree = renderer.create(
        <PlaceCardList
          offers={Offers}
          onCardTitleClick={() => {}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
