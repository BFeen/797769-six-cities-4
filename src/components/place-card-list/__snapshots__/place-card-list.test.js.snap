// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCardList snapshot checking PlaceCardList rendering in Favorites 1`] = `
<div
  className="places__list favorites__places"
>
  <article
    className="favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/0"
        onClick={[Function]}
      >
        <img
          alt="Beautiful & luxurious apartment at great location"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info favorites__card-info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/0"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Wood and stone place"
          className="place-card__image"
          height="200"
          src="img/apartment-03.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info favorites__card-info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            170
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Canal view Princengracht"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info favorites__card-info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            70
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Canal view Princengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="favorites__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        onClick={[Function]}
      >
        <img
          alt="Nice, cozy, warm big bed apartment"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info favorites__card-info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            150
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
</div>
`;

exports[`PlaceCardList snapshot checking PlaceCardList rendering in Main 1`] = `
<div
  className="places__list cities__places-list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/0"
        onClick={[Function]}
      >
        <img
          alt="Beautiful & luxurious apartment at great location"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/0"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Wood and stone place"
          className="place-card__image"
          height="200"
          src="img/apartment-03.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            170
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Canal view Princengracht"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            70
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Canal view Princengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        onClick={[Function]}
      >
        <img
          alt="Nice, cozy, warm big bed apartment"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            150
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
</div>
`;

exports[`PlaceCardList snapshot checking PlaceCardList rendering in PlaceDetails 1`] = `
<div
  className="places__list near-places__list"
>
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/0"
        onClick={[Function]}
      >
        <img
          alt="Beautiful & luxurious apartment at great location"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/0"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Wood and stone place"
          className="place-card__image"
          height="200"
          src="img/apartment-03.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            170
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Canal view Princengracht"
          className="place-card__image"
          height="200"
          src="img/room.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            70
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          Canal view Princengracht
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/3"
        onClick={[Function]}
      >
        <img
          alt="Nice, cozy, warm big bed apartment"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info "
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            150
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/3"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
</div>
`;
